### ISSUE TITLE
Row numbers

### ISSUE THREAD

**Issue by ryanwjackson** [tilo/smarter_csv#130]
Thanks Tilo, this gem is awesome!

We have a use case where we need to store and report the row number from the original CSV so it can be referenced in file if needed.  Perhaps this exists in v1.2 (though I couldn't find it), but figured I'd throw it out as a feature request in v2.0 if not.

**Comment by tilo** [comment]
yes, this would also be very useful for error reporting

**Comment by KevinColemanInc** [comment]
@ryanwjackson 

```
    @rows = SmarterCSV.process(csv, options)
    @rows.each_with_index { |r, i| r[:line_num] = i + 2 } # +1 for header line; +1 for arrays start at 0
```

I do this to tack on row numbers

**Comment by tilo** [comment]
I can add this to the header processing, so that you can get the row numbers for each of the header fields

**Comment by tilo** [comment]
@ryanwjackson @KevinColemanInc line numbers are now available in 1.7.2

new option `:with_line_numbers` will add `:csv_line_number` to each data hash
